### PR TITLE
Modify goclean to use gometalint like btcd

### PR DESCRIPTION
**Pull request by jcvernaleo**

### CODE REVIEW

File: goclean.sh [hunk 1]
```diff
@@ -5,14 +5,19 @@
 # 3. golint        (https://github.com/golang/lint)
 # 4. go vet        (http://golang.org/cmd/vet)
 # 5. test coverage (http://blog.golang.org/cover)
+#
+# gometaling (github.com/alecthomas/gometalinter) is used to run each each
```
Comment (davecgh) [inline goclean.sh]: gometalint

Comment (jcvernaleo) [inline goclean.sh]: Fixed that typo.

File: goclean.sh [hunk 2]
```diff
@@ -5,14 +5,19 @@
...
-test -z "$(gofmt -l -w .     | tee /dev/stderr)"
-test -z "$(goimports -l -w . | tee /dev/stderr)"
-test -z "$(golint .          | tee /dev/stderr)"
-go vet ./...
+test -z "$(gometalinter --disable-all \
+--enable=gofmt \
+--enable=golint \
+--enable=vet \
+--enable=goimports \
+--deadline=120s ./... | grep -v 'ALL_CAPS\|OP_\|ExampleNew' 2>&1 | tee /dev/stderr)"
```
Comment (dajohi) [inline goclean.sh]: The grep should be removed.

Comment (jcvernaleo) [inline goclean.sh]: I removed the ALL_CAPS and OP_ though since those were specific to part of btcd.

**Comment by dajohi** [comment]
OK

File: goclean.sh [hunk 3]
```diff
@@ -5,14 +5,19 @@
...
 # Automatic checks
-test -z "$(gofmt -l -w .     | tee /dev/stderr)"
-test -z "$(goimports -l -w . | tee /dev/stderr)"
-test -z "$(golint .          | tee /dev/stderr)"
-go vet ./...
+test -z "$(gometalinter --disable-all \
+--enable=gofmt \
+--enable=golint \
+--enable=vet \
+--enable=goimports \
```
Comment (davecgh) [inline goclean.sh]: Minor nitpick:

This should be in the same order as the comments above.

Comment (davecgh) [inline goclean.sh]: More specifically:

```
--enable=gofmt \
--enable=goimports \
--enable=golint \
--enable=vet \
```

Comment (jcvernaleo) [inline goclean.sh]: Fixed (and I agree, the order should match).

**Pull request merged**